test/integration/auth_sql/serverspec/auth_sql_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+
+describe 'osl-imap::auth_sql' do
+  it_behaves_like 'dovecot'
+end
+
+describe file '/etc/dovecot/conf.d/10-auth.conf' do
+  its(:content) { should match(/^!include auth-sql.conf.ext$/) }
+end
+
+describe file '/etc/dovecot/conf.d/auth-sql.conf.ext' do
+  its(:content) do
+    should match %r{passdb {
+  driver = sql
+  args = /etc/dovecot/dovecot-sql.conf.ext
+}}
+  end
+end
+
+describe file '/etc/dovecot/dovecot-sql.conf.ext' do
+  its(:content) { should match(/driver = mysql/) }
+  its(:content) { should match(/connect = host=127.0.0.1 dbname=dovecot user=dovecot_user password=dovecot_pass/) }
+  its(:content) { should match(/default_pass_scheme = SHA512-CRYPT/) }
+end
+
+describe file '/var/log/maillog' do
+  its(:content) { should match %r{postfix/local.* to=<foo@localhost>, orig_to=<foo@foo\.org>,.* status=sent \(delivered to mailbox\)$} }
+end
```

Comment:
Ideally, I'd like to be able to this using the expect script method you used in the default suite, however I get the following:
```
1 login foo bar
1 NO [AUTHENTICATIONFAILED] Authentication failed.
```

Any ideas why this is happening?

Reply:
It's not working because I didn't properly insert the user's password into the database (it's plaintext instead of the hash format that dovecot is expecting). Will update so the expect script can be used there.